internal/evaluator/rego.go

```diff
@@ -58,6 +61,44 @@ func registerOCIBlob() {
 	rego.RegisterBuiltin1(&decl, ociBlob)
 }
 
+func registerPURLIsValid() {
+	decl := rego.Function{
+		Name: purlIsValidName,
+		Decl: types.NewFunction(
+			types.Args(
+				types.Named("purl", types.S).Description("the PURL"),
+			),
+			types.Named("result", types.S).Description("PURL validity"),
+		),
+		// As per the documentation, enable memoization to ensure function evaluation is
+		// deterministic. But also mark it as non-deterministic because it does rely on external
+		// entities, i.e. OCI registry. https://www.openpolicyagent.org/docs/latest/extensions/
+		Memoize:          true,
+		Nondeterministic: true,
```

Comment:
I don't think we want this to be `Nondeterministic`. It's just a string parsing method.
(Let's also update the comment above.)

Same for `registerPURLParse`.